### ISSUE TITLE
inconsistence between spin application get & save

### ISSUE THREAD

**Issue by yinanfang** [spinnaker/spin#109]
`spin application get xxx` gives
```
{
  "attributes": {
    "cloudProviders": "kubernetes",
    "email": "xxx@example.com",
    ...
  },
  "name": "xxx"
}
```
however, the same json cannot be use for `spin application save xxx` because "email" & "cloudProviders" are not on the root level but "attributes" level. This is caused by a check in https://github.com/spinnaker/spin/blob/master/cmd/application/save.go#L88

**Comment by jtk54** [comment]
Aside from `spin`-side validation, will the application save command accept `cloudProviders` and `email` nested under `attributes`? Otherwise that seems like a strange API behavior.

**Comment by yinanfang** [comment]
not sure about whether it'll accept that or not because it failed on the client side because of spin. But if I copied "email" & "cloudProviders" to the root level, the server pipelines are updated. Also, modifying "email" & "cloudProviders" in the "attributes" doesn't cause pipeline changes but modifying them in the root level does it. I guess the server is parsing them from the root level

**Comment by jtk54** [comment]
OK I figured out why that's the case -- `spin app get <>` returns a payload that contains `attributes` which are the application attributes as well as a `clusters` which gives some information about the clusters in the application, which is useful from an UI perspective. I fixed this up so that the `app get` command acts more like you'd expect -- by default returns the app attributes, but can expand the payload if you pass a flag.

https://github.com/spinnaker/spin/pull/112.
